### PR TITLE
Enable/rollout "Getting started tour for visitors" on dotcom mode

### PR DESCRIPTION
**Pull request by erzhtor**
Closes https://github.com/sourcegraph/sourcegraph/issues/27177
Closes https://github.com/sourcegraph/sourcegraph/issues/30258

> Note: also cleans up concluded "search notebook examples" experiment https://github.com/sourcegraph/sourcegraph/issues/23096

## Description
This PR:
- Enables "Getting started tour for visitors" (previously was in [AB test](https://github.com/sourcegraph/sourcegraph/issues/30258)) on dotcom mode
- Renames "onboarding tour" prefix to "getting started tour" to avoid confusion with "search onboarding tour"
- Removes old/unused search/notebook examples from LoggedOutHomePage

## Test plan

- `SOURCEGRAPH_API_URL=https://sourcegraph.com sg start web-standalone`
- Open home page as not signed in user
- Check that "Getting Started" section is there and other "search onboarding tour" is not shown up
- Sign in and check that "getting started" section doesn't appear

## Follow-up
Delete "getting-started-tour" [feature flag](https://docs.sourcegraph.com/dev/how-to/use_feature_flags#disable-or-delete-the-feature-flag)

### CODE REVIEW

File: client/web/src/onboarding-tour/OnboardingTourManager.tsx [hunk 1]
```diff
@@ -93,6 +93,11 @@ const TourComplete: React.FunctionComponent<TelemetryProps> = ({ telemetryServic
         logTourEvent('TourGetStartedClicked')
     }, [logTourEvent])
 
+    const onRestart = useCallback(() => {
+        logTourEvent('TourRestartClicked')
```
Comment (erzhtor) [inline client/web/src/onboarding-tour/OnboardingTourManager.tsx:97]: FYI, added an event log when the "Getting started tour for visitors" Restart button clicked.

File: client/web/src/onboarding-tour/OnboardingTourManager.tsx [hunk 2]
```diff
@@ -131,14 +138,32 @@ function useSteps(telemetryService: TelemetryService): OnboardingTourStepItem[]
...
             logTourEvent('TourComplete')
             complete()
         }
     }, [complete, status, steps, logTourEvent])
 
-    return steps
+    // Handle on initial view
+    useEffect(() => {
+        if (status !== 'closed') {
+            logTourEvent('TourShown')
```
Comment (erzhtor) [inline client/web/src/onboarding-tour/OnboardingTourManager.tsx:155]: FYI, added an event log when the "Getting started tour for visitors" is rendered/shown.

File: client/web/src/onboarding-tour/OnboardingTourManager.tsx [hunk 3]
```diff
@@ -131,14 +138,32 @@ function useSteps(telemetryService: TelemetryService): OnboardingTourStepItem[]
...
+        }
+        // NOTE: intentionally excluding status to fire only once
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [])
+
+    const onClose = useCallback(() => {
+        logTourEvent('TourClosed')
```
Comment (erzhtor) [inline client/web/src/onboarding-tour/OnboardingTourManager.tsx:162]: FYI, added an event log when the "Getting started tour for visitors" is closed

**Review by tjkandala — APPROVED** [review]
Code LGTM!

**Review by ryankscott — APPROVED** [review]
LGTM

**Comment by limitedmage** [comment]
I thought the checklist experiment [was not giving positive outcomes in the A/B test](https://docs.google.com/document/d/1np4b_dz3XJvc_ciewqbgd3xFuCHb7RFhhg1L5q1CBBk/edit#bookmark=id.qhobwjvg00hr), why is it being enabled? Or am I mistaken in the metrics?

**Comment by erzhtor** [comment]
Great question, @limitedmage. There were positive signals that users would sign up (however, numbers were low to be conclusive). We will continue optimizing this onboarding tour for users to interact w/ it. 
@rrhyne and @ryankscott can elaborate more, I think.

**Pull request merged**